cmd/agent/gui/systray/Sources/gui.swift

```diff
@@ -129,8 +130,13 @@ class AgentGUI: NSObject, NSUserInterfaceValidations {
         loginItem.title = loginStatus! ? loginStatusDisableTitle : loginStatusEnableTitle
     }
 
+    func setLoginItemState(state: Bool) {
+        self.loginItem.isEnabled = state
+    }
+
     @objc func loginAction(_ sender: Any?) {
         self.loginStatus = AgentManager.switchLoginStatus()
+        setLoginItemState(state: AgentManager.checkCurrentInstallationMode())
         updateLoginItem()
```

Comment:
What exactly is the `LoginItem` in this context? my understanding is this is updating the title of an alert toggle for "Enable/Disable at login", is that correct?

per my [other comment], `AgentManager.checkCurrentInstallationMode()` will always return true so will toggling this not update the alert text?